contracts/vaults/TreasuryVault/RibbonTreasuryVaultWithAirSwap.sol

```diff
@@ -0,0 +1,1165 @@
+            "Overflow nextOptionReady"
+        );
+        optionState.nextOptionReadyAt = uint32(nextOptionReady);
+
+        _closeShort(oldOption);
+
+        // In case chargeAndDistribute was not called last round, call
+        // the function to conclude last round's performance fee and distribution
+        if (IERC20(USDC).balanceOf(address(this)) > MIN_DUST_AMOUNT) {
+            _chargeAndDistribute();
+        }
+    }
+
+    /**
+     * @notice Closes the existing short position for the vault.
+     */
+    function _closeShort(address oldOption) private {
+        uint256 lockedAmount = vaultState.lockedAmount;
+        if (oldOption != address(0)) {
+            vaultState.lastLockedAmount = uint104(lockedAmount);
+        }
+        vaultState.lockedAmount = 0;
+
+        optionState.currentOption = address(0);
+
+        if (oldOption != address(0)) {
+            uint256 withdrawAmount =
+                VaultLifecycleTreasuryWithAirSwap.settleShort(GAMMA_CONTROLLER);
+            emit CloseShort(oldOption, withdrawAmount, msg.sender);
+        }
+    }
+
+    /**
+     * @notice Rolls the vault's funds into a new short position.
+     */
+    function rollToNextOption() external onlyKeeper nonReentrant {
+        (
+            address newOption,
+            uint256 lockedBalance,
+            uint256 queuedWithdrawAmount
+        ) = _rollToNextOption(uint256(lastQueuedWithdrawAmount));
+
+        lastQueuedWithdrawAmount = queuedWithdrawAmount;
+
+        ShareMath.assertUint104(lockedBalance);
+        vaultState.lockedAmount = uint104(lockedBalance);
+
+        emit OpenShort(newOption, lockedBalance, msg.sender);
+
+        uint256 mintAmount =
+            VaultLifecycleTreasuryWithAirSwap.createShort(
+                GAMMA_CONTROLLER,
+                MARGIN_POOL,
+                newOption,
+                lockedBalance
+            );
+
+        IERC20 optionToken = IERC20(newOption);
+        optionToken.safeApprove(address(AIRSWAP_CONTRACT), mintAmount);
+    }
+
+    /**
+     * @notice Performs a swap of `currentOption` token to `asset` token with a counterparty
+     * @param order is an AirSwap order
+     */
+    function sellOptions(Types.Order calldata order)
+        external
+        onlyKeeper
+        nonReentrant
+    {
+        require(
+            order.sender.wallet == address(this),
+            "Sender can only be vault"
+        );
+        require(
+            order.sender.token == optionState.currentOption,
+            "Can only sell currentOption"
+        );
+        require(
+            order.signer.token == USDC,
+            "Can only buy with USDC"
+        );
+
+        IAirSwap(AIRSWAP_CONTRACT).swap(order);
+        // oTokens have 8 decimals, get the whole number oTokens sold
+        // oToken premium set will be in USDC and 6 decimals as well
+        uint256 oTokenPremiumToSet = order.signer.amount.div((order.sender.amount.div(10**8)));
```

Comment:
Could I get a sanity check on this calculation? The aim is to have this set the premium per oToken and thus should be 6 decimals

Reply:
From what I understood (correct me if I'm wrong):

- `order.signer.amount` has 6 decimals | because there is require above stating it must be USDC which has 6 decimals
- `order.sender.amount` has 8 decimals | because it's the oToken

I would suggest instead:
- `uint256 oTokenPremiumToSet = order.signer.amount.mul(10**8).div((order.sender.amount));`

Reply:
Fixed in [a30c9d0](https://github.com/ribbon-finance/ribbon-v2/pull/297/commits/a30c9d05c95b9d301e7c16500455108d96450e59)